Added custom ca chain to deployments

b1schumacher opened the pull request:
<!--
Thanks for submitting a change to ownCloud!

This is the bug tracker for the oCIS Helm Chart.

For fixing potential security issues please see https://owncloud.org/security/

To make it possible for us to get your change reviewed and merged please carefully fill out the requested information below.

Please note that any kind of change needs to first be submitted to the master branch which holds the next version of the oCIS Helm Chart.

Please set the following labels:

- Set label "Status:Needs-Review" for review or "Status:In-Progress" if the PR still has open tasks.
- Assignment: assign to self
- Reviewers: pick at least one
-->

## Description
Added the custom CA Chain variables to the values.yaml and added the ca to services and jobs.

## Related Issue
<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->
- Fixes [ #403](https://github.com/owncloud/ocis-charts/issues/403)

## Motivation and Context
Custom CA chains are required for multiple services, which require that ocis is trusting these ca's

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
- Tested the deployment with custom CA Chains

## Screenshots (if appropriate):

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:
<!-- Tick the checkboxes when done. -->
<!-- Raise documentation ticket in https://github.com/owncloud/docs-ocis/issues -->
<!-- or create documentation PR in https://github.com/owncloud/docs ocis/tree/master/modules/ROOT/pages/deployment/container>
- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation generated (`make docs`) and committed
- [ ] Documentation ticket raised: <link>
- [x] Documentation PR created: [<link>](https://github.com/owncloud/docs-ocis/pull/871)

d7oc commented on charts/ocis/templates/antivirus/deployment.yaml:
```diff
@@ -75,6 +80,10 @@ spec:
             - name: messaging-system-ca
               mountPath: /etc/ocis/messaging-system-ca
               readOnly: true
+            {{- if .Values.customCAChain.enabled }}
+            - name: custom-ca-chain
+              mountPath: /etc/ssl/certs/
+            {{- end }}
```
Would be good to have this fragment templated so we don't have these lines in all `deployments.yml` file but only one with a call here.

suggestion

```
{{/*
oCIS ca mount

@param .path          The path to mount
*/}}
{{- define "ocis.caPath" -}}
            {{- if .Values.customCAChain.enabled }}
            - name: custom-ca-chain
              mountPath: {{ .path }}
            {{- end }}
{{- end -}}
```

Usage afterwards with `{{- include "ocis.caPath" /etc/ssl/certs }}`

b1schumacher commented:
I changed all the entries to a template and fixed also the issue that the customCAChain was overwriting the other CA files. 

Now there are two different directories, one with the default files and one with the custom CA chain. Both are added to the SSL_CERT_DIR variable.

wkloucek commented:
@b1schumacher I applied some formatting and added enabled this feature in the linting process.

So far I didn't get the pattern for which services the custom CA has been configured / skipped. Could you maybe elaborate why it's not on all services?

Also what's missing are the CronJobs, some of them only talk to NATS but even for this, we could make use of this feature.

b1schumacher commented:
@wkloucek I added the missing services and the jobs.

wkloucek commented:
I changed this to point to `stable-5` instead of `main` which is now tracking ocis rolling releases

wkloucek commented on charts/ocis/values.yaml:
```diff
@@ -613,6 +613,15 @@ secretRefs:
+  # -- Custom CA enables SSL_CERT_DIR in pods with the additional path /etc/ssl/custom.
+  enabled: false
+  # -- If custom CA chain is enabled this attribute mounts the existing secret to /etc/ssl/custom.
+  existingSecret: ""
```
Is there a particular reason for the CA chain to be put in a secret? From what I know, a CA chain is public information.

wkloucek replied:
I changed it to a ConfigMap in https://github.com/owncloud/ocis-charts/pull/599/commits/e3443826ee3659970b262e7c767d2cac786a4c8d

d7oc reviewed (changes requested):
LGTM aside the typo. Also verified locally via k3d.

d7oc reviewed (approved):
LGTM
